spacy/pipeline/entity_linker.py

```diff
@@ -153,10 +154,17 @@ def begin_training(
 
         DOCS: https://nightly.spacy.io/api/entitylinker#begin_training
         """
-        self.require_kb()
+        self._ensure_examples(get_examples)
+        self._require_kb()
         nO = self.kb.entity_vector_length
-        self.set_output(nO)
-        self.model.initialize()
+        doc_sample = []
+        vector_sample = []
+        for example in islice(get_examples(), 10):
+            doc_sample.append(example.x)
+            vector_sample.append(self.model.ops.alloc1f(nO))
+        assert doc_sample
```

Comment:
I agree with the reasoning for regular asserts here. Maybe we could have a default error message we can show alongside it that says something like "This is likely a bug in spaCy" and then use that? For instance:

```suggestion
        assert doc_sample, Errors.EXXX
```